src/store/group-management/saga.ts

```diff
@@ -1,22 +1,47 @@
-import { SagaActionTypes, Stage, setStage } from '.';
-import { call, fork, put, race, take, delay } from 'redux-saga/effects';
+import { SagaActionTypes, Stage, setStage, setRoomMembers } from '.';
+import { Chat, chat } from '../../lib/chat';
+import { call, fork, put, race, take, select } from 'redux-saga/effects';
 import { Events, getAuthChannel } from '../authentication/channels';
+import { denormalize as denormalizeUsers } from '../users';
 
 export function* reset() {
+  yield put(setRoomMembers([]));
   yield put(setStage(Stage.None));
 }
 
+export function* roomMembersSelected(action) {
+  const { users: selectedMembers, roomId } = action.payload;
+
+  try {
+    if (!roomId || !selectedMembers) {
+      return;
+    }
+
+    const userIds = selectedMembers.map((user) => user.value);
+    const users = yield select((state) => denormalizeUsers(userIds, state));
+
+    const chatClient: Chat = yield call(chat.get);
+    yield call([chatClient, chatClient.addMembersToRoom], roomId, users);
+    yield put(setRoomMembers(selectedMembers));
+
+    return Stage.None;
+  } catch (error) {}
```

Comment:
So, if there's an error, the user just stays on the screen they're on? Should there be feedback or are we just going to assume the user will continue to click the button?

Side note: I am also a little worried that we don't disable the button while processing this. It opens us up to all the various "double click" bugs.

Reply:
I've asked Eth how we want to handle the scenario where this would error. For now I've added a console log, with a comment to handle once we have UI for it/decided how we want to handle it.

re side note: I've added loading to this now which will disable the button while processing the adding of members: `isAddingMembers`

Reply:
Nice